projects/rocprofiler-systems/source/lib/core/trace_cache/sample_type.hpp

```diff
@@ -182,6 +185,43 @@ struct pmc_event_with_sample : in_time_sample
     size_t      value;
 };
 
+struct amd_smi_sample : storage_parsed_type_base
+{
+    enum class settings_positions : uint8_t
+    {
+        busy = 0,
+        temp,
+        power,
+        mem_usage,
+        vcn_activity,
+        jpeg_activity
+    };
+
+    size_t               settings;
```

Comment:
```suggestion
    size_t               settings;   // bitfield
```
You may want to use `uint64_t`, since it's guaranteed to be a constant size and settings is used as a bitfield.